cloud/deploy/deploy.go

```diff
@@ -996,3 +1002,89 @@ func DeployClientImage(deployInput InputClientDeploy) error { //nolint:gocritic
 
 	return nil
 }
+
+// validateClientImageRuntimeVersion validates that the client image runtime version
+// is not newer than the deployment runtime version
+func validateClientImageRuntimeVersion(deployInput InputClientDeploy, platformCoreClient astroplatformcore.CoreClient) error { //nolint:gocritic
+	// Skip validation if no deployment ID provided
+	if deployInput.DeploymentID == "" {
+		return nil
+	}
+
+	// Get current context for organization info
+	c, err := config.GetCurrentContext()
+	if err != nil {
+		return errors.Wrap(err, "failed to get current context")
+	}
+
+	// Get deployment information
+	deployInfo, err := getImageName(deployInput.DeploymentID, c.Organization, platformCoreClient)
+	if err != nil {
+		return errors.Wrap(err, "failed to get deployment information")
+	}
+
+	// Parse Dockerfile.client to get client image runtime version
+	dockerfileClientPath := filepath.Join(deployInput.Path, "Dockerfile.client")
+	if _, err := os.Stat(dockerfileClientPath); os.IsNotExist(err) {
+		// If Dockerfile.client doesn't exist, skip validation
+		return nil
+	}
+
+	cmds, err := docker.ParseFile(dockerfileClientPath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse Dockerfile.client: %s", dockerfileClientPath)
+	}
+
+	baseImage := docker.GetImageFromParsedFile(cmds)
+	if baseImage == "" {
+		return errors.New("failed to find base image in Dockerfile.client")
+	}
+
+	// Extract runtime version from the base image tag
+	clientRuntimeVersion, err := extractRuntimeVersionFromImage(baseImage)
+	if err != nil {
+		// If we can't extract version, skip validation with a warning
+		fmt.Printf("Warning: Could not extract runtime version from client image %s, skipping version validation\n", baseImage)
```

Comment:
nit: if we are simply warning in case of error over here, then maybe it makes sense to do the same for other errors before this point as well.